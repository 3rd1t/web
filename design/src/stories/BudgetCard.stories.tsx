import React from "react";
import { BudgetCard } from '../components/BudgetCard';

export default {
  title: "Budget Card",
};

export const card = () => {

  return (
    <BudgetCard  expensesCount= {195} totalBudget={1500.00} start={new Date()} end={new Date()} budgeted={4.99} expenses={625.12} income={10.00}/>
  )
};
