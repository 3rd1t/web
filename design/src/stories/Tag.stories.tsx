import React from 'react';
import Tag from "../components/Tag"
export default { title: 'Tag' };

export const Default = () => {
    return (
        <Tag>Hello Tag</Tag>
    )
} 
