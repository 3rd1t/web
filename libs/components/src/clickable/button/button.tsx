import { prependListener } from "process";
import React from "react";
import { Base } from "../base/base";

/* eslint-disable-next-line */
export interface ButtonProps {
  label: string;
  textColor?: string;
  bgColor?: string;
  onClick: () => void;
  iconLeft?: React.ReactNode;
  iconRight?: React.ReactNode;
  className?: string;
}

export const Button = (props: ButtonProps) => {
  const className = [
    "focus:outline-none transition duration-200 rounded-sm py-1 px-4 font-medium",
    props.textColor,
    props.bgColor ||
      "bg-gradient-to-tr from-indigo-900 to-purple-800 hover:from-indigo-800 hover:to-purple-700",
    props.className,
  ].join(" ");

  return (
    <button className={className} onClick={props.onClick}>
      <Base
        label={props.label}
        iconLeft={props.iconLeft}
        iconRight={props.iconRight}
      ></Base>
    </button>
  );
};

export default Button;
