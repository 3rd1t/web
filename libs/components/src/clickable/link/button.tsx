import React from "react";
import { Link } from "./link";

/* eslint-disable-next-line */
export interface ButtonProps {
  label?: string;
  textColor?: string;
  bgColor?: string;
  href: string;
  iconLeft?: React.ReactNode;
  iconRight?: React.ReactNode;
  className?: string;
}

export const Button = (props: ButtonProps) => {
  return (
    <Link
      label={props.label}
      textColor={props.textColor}
      bgColor={
        props.bgColor ||
        "bg-gradient-to-t from-indigo-900 to-purple-800 hover:from-indigo-800 hover:to-purple-700"
      }
      href={props.href}
      iconLeft={props.iconLeft}
      iconRight={props.iconRight}
      className={props.className}
    ></Link>
  );
};

export default Button;
