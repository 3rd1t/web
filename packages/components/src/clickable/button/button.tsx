import React from "react"
import { Base } from "../base/base"
import { getHoverColor } from "../util/hoverColor"

/* eslint-disable-next-line */
export interface ButtonProps {
  label: string | React.ReactNode
  textColor: string
  bgColor?: string
  onClick: () => void
  iconLeft?: React.ReactNode
  iconRight?: React.ReactNode
  width?: string
}

export const Button = (props: ButtonProps) => {
  const classNames = [
    "focus:outline-none",
    props.width,
    props.textColor,
    getHoverColor(props.textColor),
    props.bgColor,
    getHoverColor(props.bgColor),
  ].join(" ")

  return (
    <button onClick={props.onClick} className={classNames}>
      <Base label={props.label} iconLeft={props.iconLeft} iconRight={props.iconRight}></Base>
    </button>
  )
}

export default Button
