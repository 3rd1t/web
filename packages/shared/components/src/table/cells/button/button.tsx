import React from "react"
import Wrapper from "../wrapper/wrapper"

/* eslint-disable-next-line */
export interface ButtonProps {
  label: string
  onClick: () => void
}

export const Button = ({ label, onClick }: ButtonProps) => {
  return (
    <Wrapper>
      <button onClick={onClick} className="text-sm font-medium leading-5 text-indigo-600 hover:text-indigo-900">
        {label}
      </button>
    </Wrapper>
  )
}

export default Button
