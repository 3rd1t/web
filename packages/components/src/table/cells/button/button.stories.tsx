import { action } from "@storybook/addon-actions"

import React from "react"
import { Button, ButtonProps } from "./button"

export default {
  component: Button,
  title: "components/table/cell/Button",
  argTypes: { onClick: { action: "clicked" } },
}

export const primary = () => {
  /* eslint-disable-next-line */
  const props: ButtonProps = {
    label: "Label",
    onClick: action("button-click"),
  }

  return <Button label={props.label} onClick={props.onClick} />
}
