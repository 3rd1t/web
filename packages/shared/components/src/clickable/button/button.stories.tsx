import React from "react"
import { Button, ButtonProps } from "./button"

export default {
  component: Button,
  title: "components/clickable/Button",
}

export const primary = () => {
  /* eslint-disable-next-line */
  const props: ButtonProps = {
    label: "Label",
    bgColor: "bg-gray-200",
    textColor: "text-gray-900",
    onClick: () => {},
  }

  return <Button label={props.label} textColor={props.textColor} bgColor={props.bgColor} onClick={props.onClick} />
}

export const black = () => {
  /* eslint-disable-next-line */
  const props: ButtonProps = {
    label: "Label",
    bgColor: "bg-carbon-900",
    textColor: "text-gray-100",
    onClick: () => {},
  }

  return <Button label={props.label} textColor={props.textColor} bgColor={props.bgColor} onClick={props.onClick} />
}

export const iconOnly = () => {
  /* eslint-disable-next-line */
  const props: ButtonProps = {
    bgColor: "bg-gray-200",
    textColor: "text-green-500",
    onClick: () => {},
    label: (
      <svg className="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M5 13l4 4L19 7" />
      </svg>
    ),
  }

  return <Button label={props.label} textColor={props.textColor} bgColor={props.bgColor} onClick={props.onClick} />
}
export const red = () => {
  const props: ButtonProps = {
    label: "Label",
    bgColor: "bg-red-600",
    textColor: "text-red-100",
    onClick: () => {},
  }

  return <Button label={props.label} textColor={props.textColor} bgColor={props.bgColor} onClick={props.onClick} />
}

export const block = () => {
  const props: ButtonProps = {
    label: "Label",
    bgColor: "bg-carbon-900",
    textColor: "text-gray-100",
    onClick: () => {},
  }

  return (
    <Button
      width="w-full"
      label={props.label}
      textColor={props.textColor}
      bgColor={props.bgColor}
      onClick={props.onClick}
    />
  )
}

export const withIconLeft = () => {
  const props: ButtonProps = {
    label: "Label",
    bgColor: "bg-carbon-900",
    textColor: "text-gray-100",
    onClick: () => {},
    iconLeft: (
      <svg className="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path d="M12 14l9-5-9-5-9 5 9 5z" />
        <path d="M12 14l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14z" />
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          d="M12 14l9-5-9-5-9 5 9 5zm0 0l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14zm-4 6v-7.5l4-2.222"
        />
      </svg>
    ),
  }

  return (
    <Button
      label={props.label}
      textColor={props.textColor}
      bgColor={props.bgColor}
      onClick={props.onClick}
      iconLeft={props.iconLeft}
    />
  )
}

export const withIconRight = () => {
  const props: ButtonProps = {
    label: "Label",
    bgColor: "bg-carbon-900",
    textColor: "text-gray-100",
    onClick: () => {},
    iconRight: (
      <svg className="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M14 5l7 7m0 0l-7 7m7-7H3" />
      </svg>
    ),
  }

  return (
    <Button
      label={props.label}
      textColor={props.textColor}
      bgColor={props.bgColor}
      onClick={props.onClick}
      iconRight={props.iconRight}
    />
  )
}

export const withIconLeftAndRight = () => {
  const props: ButtonProps = {
    label: "Label",
    bgColor: "bg-carbon-900",
    textColor: "text-gray-100",
    onClick: () => {},
    iconLeft: (
      <svg className="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          d="M7 21a4 4 0 01-4-4V5a2 2 0 012-2h4a2 2 0 012 2v12a4 4 0 01-4 4zm0 0h12a2 2 0 002-2v-4a2 2 0 00-2-2h-2.343M11 7.343l1.657-1.657a2 2 0 012.828 0l2.829 2.829a2 2 0 010 2.828l-8.486 8.485M7 17h.01"
        />
      </svg>
    ),
    iconRight: (
      <svg className="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M19 9l-7 7-7-7" />
      </svg>
    ),
  }

  return (
    <Button
      label={props.label}
      textColor={props.textColor}
      bgColor={props.bgColor}
      onClick={props.onClick}
      iconLeft={props.iconLeft}
      iconRight={props.iconRight}
    />
  )
}
