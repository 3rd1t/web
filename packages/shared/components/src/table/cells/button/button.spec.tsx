import { render } from "@testing-library/react"
import * as stories from "./button.stories"

describe("Button", () => {
  describe("Stories", () => {
    Object.entries(stories).forEach((s) => {
      if (s[0] !== "default") {
        it(s[0] + "should match snapshot", () => {
          const { baseElement } = render(s[1]())
          expect(baseElement).toMatchSnapshot()
        })
      }
    })
  })
})
