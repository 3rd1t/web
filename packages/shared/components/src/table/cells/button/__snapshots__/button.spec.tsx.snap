// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Stories primaryshould match snapshot 1`] = `
<body>
  <div>
    <div
      class="px-6 py-4 whitespace-no-wrap"
    >
      <button
        class="text-sm font-medium leading-5 text-indigo-600 hover:text-indigo-900"
      >
        Label
      </button>
    </div>
  </div>
</body>
`;
